feat: prepare-complex, merge for TRCs, nn-xTB module, EXESS export keywords, EXESS opimization with residues param

seankhl opened the pull request:
Numerous improvements.

seankhl committed:
feat: support nnxtb

seankhl committed:
feat: frag_keywords can be None / null (no fragmentation)

seankhl committed:
fix: correctly name rex functions

seankhl committed:
feat: implement prepare-complex

seankhl committed:
feat: PDB input and save_outputs for prepare-protein

seankhl committed:
feat: param to save & extract bins in save_object

seankhl committed:
feat: implement merge_trcs & various mol fixes

seankhl committed:
chore: add nnxtb to prod

seankhl committed:
fix: correctly consdtruct all-None fragment case

Also, implement save_energy_outputs.

seankhl committed:
feat: implement exess export keywords

seankhl committed:
feat: pass residues to opt to support MM

qweiliang1 commented on src/rush_py2/prepare_protein.py:
`capping_style="always"` seems to be broken. I cant use it anymore  

```
  23       (prepare_protein_rex::PrepareProteinOptions {
  24         ph = None,
  25         naming_scheme = None,
  26         capping_style = always,
  27         truncation_threshold = None,
  28       })
  29       [( (obj_j topology), (obj_j residues), (obj_j chains) )]
  30 in
  31   prepare_protein "51234ad29caa9c85892a8f70d4da84dcf1d20030af3a17c09dde2b73b19ea6cb" "b5d7f13f71034854f2ef0c8c23b508d91f35c1c8159d1fc47aefdf86c01a6fbb" "bcbb7f34602c3e6611de90deabe82bddb376128e8ec7f3cca2abfb714406c136"

Errors:
26:25 - 26:31: Unbound variable: always
```

seankhl replied:
Fixed this. Tested each param with various args, I think all are working.

qweiliang1 commented on src/rush_py2/prepare_complex.py:
prepare_complex seem to still be saving the outputs onto my cwd. Also, is having T, R and C seperate a deliberate design choice? It's less intuitive for a comp chemist, because we hardly view them seperately

seankhl replied:
1. Oops yeah will fix the outut location problem
2. Right now I'm having it try to copy the behavior of prepare_protein, which is like that for tengu module related reasons (has to do with how the TRC parts are stored in the cloud object store, can go into more detail if you want). I totally agree it's less intuitive and a pain to work with it this way.

seankhl replied:
Should be resolved, prepare_complex just returns the whole TRC in memory now. You can choose to save it or not.

qweiliang1 commented on src/rush_py2/trc/merge.py:
Ive tried multiple ways of running this, but none of it works

seankhl replied:
Wrote a test in `test_merge.py` that shows the intended usage, let me know if there are any changes you think would be good to make to the interface, what cases it can handle, etc.

seankhl committed:
fix: improvements to prepare_complex

- allow passing PDB or TRC to prepare_complex
- don't write temporary files to current dir
- return a TRC object directly in memory, not paths to T, R, C files

seankhl committed:
fix: correctly pass args to prepare_protein

seankhl committed:
feat: write test for merge_trcs

seankhl committed:
fix: don't skip validation; remove unnecessary file script

Commit:
fix: prepare protein to pass

Commit:
feat: use pytest

seankhl commented:
Gonna call this good to go
